File: payView/DWPayView/DWPayView.m
//
//  DWPayView.m
//  payView
//
//  Created by DA WENG on 2017/1/9.
//  Copyright © 2017年 DA WENG. All rights reserved.
//

#import "DWPayView.h"
#import "DWEnterPasswordView.h"
#import "DWBankListView.h"
#import "DWSizeDefined.h"
#import "DWCrossView.h"
#import "DWArrowView.h"
#import "DWManageData.h"
#import "DWViewConvertToImage.h"
@interface DWPayView ()<UITableViewDelegate,UITableViewDataSource,ManageDataDelegate>
@property(nonatomic,strong)NSArray* titleArray;
@property(nonatomic,strong)DWManageData* manageData;
@end

@implementation DWPayView
{
    UIScrollView* scrollView;
    UITableView* payMethodTableView;
    DWEnterPasswordView* enterPasswordView;
    DWBankListView* bankListView;
    UILabel* secondTitle;
    NSDictionary* payMethodDict;
    UIButton* backBtn;
    NSString* orderFeeAmount;
}
//- (instancetype)initWithFrame:(CGRect)frame {
//    if (self = [super initWithFrame:frame]) {
//        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(disMissView) name:@"finishPay" object:nil];
//        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(disableBackBtn) name:@"disableBackBtn" object:nil];
//        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(bankCardSelected:) name:@"bankCardSelected" object:nil];
////        [self initContent];
//    }
//    return self;
//}


-(instancetype)initWithorderFee:(NSString*)orderFee whomToPay:(NSString*)whomToPay{
    
    self=[super init];
    
    if (self) {
        NSLog(@"%@",orderFee);
        //        orderFeeAmount = orderFee;
        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(disMissView) name:@"finishPay" object:nil];
        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(disableBackBtn) name:@"disableBackBtn" object:nil];
        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(bankCardSelected:) name:@"bankCardSelected" object:nil];
        [self initContentWithOrderFee:orderFee whomToPay:whomToPay];
        
    }
    
    return self;
    
}

- (void)initContentWithOrderFee:(NSString*)orderFee whomToPay:(NSString*)whomToPay
{
    
    DWManageData* manageData = [[DWManageData alloc]init];
    manageData.delegate = self;
    [manageData getInitPayMethod];
    self.frame = CGRectMake(0, 0, dwDEVICESCREENWIDTH, dwDEVICESCREENHEIGHT);
    self.backgroundColor = [UIColor colorWithRed:0 green:0 blue:0 alpha:0.2];
    //    self.userInteractionEnabled = YES;
    //    [self addGestureRecognizer:[[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(disMissView)]];
    
    self.titleArray = @[@"支付方式",@"持卡人"];
    
    if(scrollView ==nil){
        
        scrollView = [[UIScrollView alloc]initWithFrame:CGRectMake(0, dwDEVICESCREENHEIGHT-dwSCROLLVIEWHEIGHT,dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
        scrollView.contentSize = CGSizeMake(dwDEVICESCREENWIDTH*2, dwSCROLLVIEWHEIGHT);
        scrollView.backgroundColor = [UIColor whiteColor];
        [self addSubview:scrollView];
        [scrollView setBounces:NO];
        [scrollView setShowsHorizontalScrollIndicator:NO];
        [scrollView setShowsVerticalScrollIndicator:NO];
        scrollView.scrollEnabled = NO;
        
        UIView* firstTitleView = [[UIView alloc]initWithFrame:CGRectMake(0, 0, dwDEVICESCREENWIDTH, dwBUTTONWIDTH)];
        UILabel* firstTitle = [[UILabel alloc]initWithFrame:CGRectMake(0, 0, firstTitleView.frame.size.width, firstTitleView.frame.size.height)];
        firstTitle.text = @"鸡年大吉吧支付";
        firstTitle.textAlignment = NSTextAlignmentCenter;
        [firstTitleView addSubview:firstTitle];
        [scrollView addSubview:firstTitleView];
        
        UIView* secondTitleView = [[UIView alloc]initWithFrame:CGRectMake(dwDEVICESCREENWIDTH, 0, dwDEVICESCREENWIDTH, dwBUTTONWIDTH)];
        secondTitle = [[UILabel alloc]initWithFrame:CGRectMake(0, 0, secondTitleView.frame.size.width, secondTitleView.frame.size.height)];
        secondTitle.textAlignment = NSTextAlignmentCenter;
        [secondTitleView addSubview:secondTitle];
        [scrollView addSubview:secondTitleView];
        
        
        UIButton* closeBtn = [[UIButton alloc]initWithFrame:CGRectMake(0, 0, dwBUTTONWIDTH, dwBUTTONWIDTH)];
        DWCrossView* crossView = [[DWCrossView alloc]initWithFrame:closeBtn.frame];
        [scrollView addSubview:crossView];
        UIImage * crossimage = [[DWViewConvertToImage alloc]initWithView:crossView];
        [closeBtn setImage:crossimage forState:UIControlStateNormal];
        [closeBtn addTarget:self action:@selector(disMissView) forControlEvents:UIControlEventTouchUpInside];
        [scrollView addSubview:closeBtn];
        
        
        UIButton* nextBtn = [[UIButton alloc]initWithFrame:CGRectMake(dwMARGINTOLEFT, scrollView.frame.size.height-dwBUTTONWIDTH*1.1-dwMARGINTOLEFT*1.5, dwDEVICESCREENWIDTH-dwMARGINTOLEFT*2, dwBUTTONWIDTH*1.1)];
        nextBtn.layer.masksToBounds = YES;
        nextBtn.layer.cornerRadius = 5;
        nextBtn.backgroundColor = [UIColor colorWithRed:71.0/255.0 green:37.0/255.0 blue:116.0/255.0 alpha:1];
        [nextBtn setTitle:@"确认付款" forState:UIControlStateNormal];
        [nextBtn addTarget:self action:@selector(nextClick) forControlEvents:UIControlEventTouchUpInside];
        [scrollView addSubview:nextBtn];
        
        
        backBtn = [[UIButton alloc]initWithFrame:CGRectMake(dwDEVICESCREENWIDTH, 0, dwBUTTONWIDTH, dwBUTTONWIDTH)];
        DWArrowView* arrowView = [[DWArrowView alloc]initWithFrame:backBtn.frame];
        [scrollView addSubview:arrowView];
        UIImage * arrowimage = [[DWViewConvertToImage alloc]initWithView:arrowView];
        [backBtn setImage:arrowimage forState:UIControlStateNormal];
        [backBtn addTarget:self action:@selector(backClick) forControlEvents:UIControlEventTouchUpInside];
        [scrollView addSubview:backBtn];
        
        
        UILabel* lineLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, closeBtn.frame.size.height, dwDEVICESCREENWIDTH, 0.2)];
        lineLabel.backgroundColor = [UIColor lightGrayColor];
        [scrollView addSubview:lineLabel];
        
        UILabel* payToWhomLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, closeBtn.frame.size.height*2, dwDEVICESCREENWIDTH, closeBtn.frame.size.height/2)];
        payToWhomLabel.text = [NSString stringWithFormat:@"向%@支付",whomToPay];
        payToWhomLabel.textAlignment = NSTextAlignmentCenter;
        [scrollView addSubview:payToWhomLabel];
        
        UILabel* moneyLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, payToWhomLabel.frame.origin.y+payToWhomLabel.frame.size.height, dwDEVICESCREENWIDTH, payToWhomLabel.frame.size.height*2.3)];
        moneyLabel.text =orderFee;
        moneyLabel.font = [UIFont systemFontOfSize:36];
        moneyLabel.textAlignment =NSTextAlignmentCenter;
        [scrollView addSubview:moneyLabel];
        
        payMethodTableView = [[UITableView alloc]initWithFrame:CGRectMake(0, moneyLabel.frame.origin.y+moneyLabel.frame.size.height+payToWhomLabel.frame.size.height,dwDEVICESCREENWIDTH , dwBUTTONWIDTH*2)];
        payMethodTableView.rowHeight = dwBUTTONWIDTH;
        payMethodTableView.dataSource = self;
        payMethodTableView.delegate = self;
        payMethodTableView.bounces = NO;
        [scrollView addSubview:payMethodTableView];
        
        UILabel* lineLabe2 = [[UILabel alloc]initWithFrame:CGRectMake(15, payMethodTableView.frame.origin.y, dwDEVICESCREENWIDTH, 0.2)];
        lineLabe2.backgroundColor = [UIColor lightGrayColor];
        [scrollView addSubview:lineLabe2];
        
    }
    
}

#pragma mark tableView

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 2;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    static NSString *cellIdentifier = @"cellIdentifier";
    
    UITableViewCell *cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleValue1 reuseIdentifier:cellIdentifier];
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    
    
    cell.textLabel.text = [self.titleArray objectAtIndex:[indexPath row]];
    
    
    
    if (indexPath.row == 0) {
        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
        
        cell.detailTextLabel.text = [payMethodDict objectForKey:@"bankCard"];//根据获取数据修改
        
    }else if (indexPath.row == 1){
        
        cell.detailTextLabel.text = [payMethodDict objectForKey:@"userName"];//根据获取数据修改
        
    }
    
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    if (indexPath.row == 0) {
        [self bankList];
    }
}


- (void)showInView:(UIView *)view
{
    if (!view)
    {
        return;
    }
    
    [view addSubview:self];
    [view addSubview:scrollView];
    
    [scrollView setFrame:CGRectMake(0, dwDEVICESCREENHEIGHT, dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
    
    [UIView animateWithDuration:0.3 animations:^{
        
        self.alpha = 1.0;
        backBtn.enabled = YES;
        [scrollView setFrame:CGRectMake(0, dwDEVICESCREENHEIGHT - dwSCROLLVIEWHEIGHT, dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
        
    } completion:nil];
}

//移除从上向底部弹下去的UIView（包含遮罩）
- (void)disMissView
{
    [scrollView setFrame:CGRectMake(0, dwDEVICESCREENHEIGHT - dwSCROLLVIEWHEIGHT, dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
    [UIView animateWithDuration:0.3f
                     animations:^{
                         
                         self.alpha = 0.0;
                         
                         [scrollView setFrame:CGRectMake(0, dwDEVICESCREENHEIGHT, dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
                     }
                     completion:^(BOOL finished){
                         [scrollView setContentOffset:CGPointMake(0, 0) animated:NO];
                         [self removeFromSuperview];
                         [scrollView removeFromSuperview];
                         
                     }];
    
}

- (UIImage *)getImageFromView:(UIView *)view {
    UIGraphicsBeginImageContextWithOptions(view.bounds.size, NO, [UIScreen mainScreen].scale);
    [view.layer renderInContext:UIGraphicsGetCurrentContext()];
    UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return image;
}



-(void)nextClick{
    CGFloat offsetX = scrollView.bounds.size.width;
    [scrollView setContentOffset:CGPointMake(offsetX, 0) animated:YES];
    secondTitle.text = @"输入数字支付密码";
    enterPasswordView = [[DWEnterPasswordView alloc]initWithFrame:CGRectMake(dwDEVICESCREENWIDTH, dwBUTTONWIDTH, dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
    [scrollView addSubview:enterPasswordView];
    
    
}

-(void)backClick{
    
    [scrollView setContentOffset:CGPointMake(0, 0) animated:YES];
    [enterPasswordView.textField resignFirstResponder];
    
}

-(void)bankList{
    CGFloat offsetX = scrollView.bounds.size.width;
    [scrollView setContentOffset:CGPointMake(offsetX, 0) animated:YES];
    secondTitle.text = @"选择支付方式";
    bankListView = [[DWBankListView alloc]initWithFrame:CGRectMake(dwDEVICESCREENWIDTH, dwBUTTONWIDTH, dwDEVICESCREENWIDTH, dwSCROLLVIEWHEIGHT)];
    [scrollView addSubview:bankListView];
    
}


//代理传获取到的默认银行卡信息
-(void)passData:(id)data{
    NSLog(@"payview%@",data);
    NSDictionary* dict = [[NSDictionary alloc]init];
    dict = data;
    
    payMethodDict = dict;
    [payMethodTableView reloadData];
}


//输入完密码后禁用返回键
-(void)disableBackBtn{
    
    backBtn.enabled = NO;
    backBtn.adjustsImageWhenDisabled = NO;
    
}

//回传选号的银行卡信息
-(void)bankCardSelected:(NSNotification *)notification{
    [self backClick];
    payMethodDict = [notification userInfo];
    NSLog(@"%@",payMethodDict);
    [payMethodTableView reloadData];
    
    
}

@end

File: DWPayViewSwift/DWPayViewSwift/DWPayView.swift
//
//  DWPayView.swift
//  DWPayViewSwift
//
//  Created by DA WENG on 2017/2/20.
//  Copyright © 2017年 DA WENG. All rights reserved.
//

import UIKit

class DWPayView: UIView ,UITableViewDelegate,UITableViewDataSource,DWManageDataDelegate{
    
    var payMethodDict = NSDictionary()
    let titleArray = ["支付方式","持卡人"]
    var scrollView = UIScrollView()
    var secondTitle = UILabel()
    var payMethodTableView = UITableView()
    var backBtn = UIButton()
    var enterPasswordView = DWEnterPasswordView()
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
    }
    
    convenience init(orderFee:String,whomToPay:String) {
        
        self.init()

        initContent(orderFee: orderFee, whomToPay: whomToPay)
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    
    func initContent(orderFee:String,whomToPay:String) {
        self.frame = CGRect(x: 0, y: 0, width: dwDEVICESCREENWIDTH, height: dwDEVICESCREENHEIGHT)
        self.backgroundColor = UIColor.init(colorLiteralRed: 0, green: 0, blue: 0, alpha: 0.2)
        
        
        scrollView = UIScrollView(frame: CGRect(x: 0, y: dwDEVICESCREENHEIGHT-dwSCROLLVIEWHEIGHT, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT))
        scrollView.contentSize = CGSize(width: dwDEVICESCREENWIDTH*2, height: dwSCROLLVIEWHEIGHT)
        scrollView.backgroundColor = UIColor.white
        scrollView.bounces = false
        scrollView.showsVerticalScrollIndicator = false
        scrollView.showsHorizontalScrollIndicator = false
        scrollView.isScrollEnabled = false
        self.addSubview(scrollView)
        
        let firstTitleView = UIView(frame: CGRect(x: 0, y: 0, width: dwDEVICESCREENWIDTH, height: dwBUTTONWIDTH))
        let firstTitle = UILabel(frame: CGRect(x: 0, y: 0, width: firstTitleView.frame.size.width, height:  firstTitleView.frame.size.height))
        firstTitle.text = "安全支付"
        firstTitle.textAlignment = NSTextAlignment.center
        firstTitleView .addSubview(firstTitle)
        scrollView.addSubview(firstTitleView)
        
        let secondTitleView = UIView(frame: CGRect(x: dwDEVICESCREENWIDTH, y: 0, width: dwDEVICESCREENWIDTH, height: dwBUTTONWIDTH))
        secondTitle = UILabel(frame: CGRect(x: 0, y: 0, width: secondTitleView.frame.size.width, height: secondTitleView.frame.size.height))
        secondTitle.textAlignment = NSTextAlignment.center
        secondTitleView.addSubview(secondTitle)
        scrollView.addSubview(secondTitleView)
        
        let closeBtn = UIButton(frame: CGRect(x: 0, y: 0, width: dwBUTTONWIDTH, height: dwBUTTONWIDTH))
        let crossView = DWCrossView(frame: closeBtn.frame)
        scrollView.addSubview(crossView)
        let crossImage = DWViewConvertToImage.getImageFromView(view: crossView)
        closeBtn.setImage(crossImage, for: UIControlState.normal)
        closeBtn.backgroundColor = UIColor.red
        closeBtn.addTarget(self, action:#selector(DWPayView.disMissView) , for:UIControlEvents.touchUpInside)
        scrollView.addSubview(closeBtn)
        
        let nextBtn = UIButton(frame: CGRect(x: dwMARGINTOLEFT, y: scrollView.frame.size.height-dwBUTTONWIDTH*1.1-dwMARGINTOLEFT*1.5, width: dwDEVICESCREENWIDTH-dwMARGINTOLEFT*2, height: dwBUTTONWIDTH*1.1))
        nextBtn.layer.masksToBounds = false
        nextBtn.layer.cornerRadius = 5
        nextBtn.backgroundColor = UIColor.init(red: 71.0/255.0, green: 37.0/255.0, blue: 116.0/255.0, alpha: 1)
        nextBtn.setTitle("确认付款", for: UIControlState.normal)
        nextBtn.addTarget(self, action: #selector(nextClick), for: UIControlEvents.touchUpInside)
        scrollView.addSubview(nextBtn)
        
        backBtn = UIButton(frame: CGRect(x: dwDEVICESCREENWIDTH, y: 0, width: dwBUTTONWIDTH, height: dwBUTTONWIDTH))
        let arrowView = DWArrowView(frame: backBtn.frame)
        scrollView.addSubview(arrowView)
        let arrowimage = DWViewConvertToImage.getImageFromView(view: arrowView)
        backBtn.setImage(arrowimage, for: UIControlState.normal)
        backBtn.addTarget(self, action: #selector(backClick), for: UIControlEvents.touchUpInside)
        scrollView.addSubview(backBtn)
        
        let lineLabel = UILabel(frame: CGRect(x: 0, y: closeBtn.frame.size.height, width: dwDEVICESCREENWIDTH, height: 0.2))
        lineLabel.backgroundColor = UIColor.lightGray
        scrollView.addSubview(lineLabel)
        
        let payToWhomLabel = UILabel(frame: CGRect(x: 0, y: closeBtn.frame.size.height*2, width: dwDEVICESCREENWIDTH, height: closeBtn.frame.size.height/2))
        payToWhomLabel.text = "向\(whomToPay)支付"
        payToWhomLabel.textAlignment = NSTextAlignment.center
        scrollView.addSubview(payToWhomLabel)
        
        
        let moneyLabel = UILabel(frame: CGRect(x: 0, y: payToWhomLabel.frame.origin.y+payToWhomLabel.frame.size.height, width: dwDEVICESCREENWIDTH, height: payToWhomLabel.frame.size.height*2.3))
        moneyLabel.text = orderFee
        moneyLabel.font = UIFont.systemFont(ofSize: 36)
        moneyLabel.textAlignment = NSTextAlignment.center
        scrollView.addSubview(moneyLabel)
        
        
        payMethodTableView = UITableView(frame: CGRect(x: 0, y: moneyLabel.frame.origin.y+moneyLabel.frame.size.height+payToWhomLabel.frame.size.height, width: dwDEVICESCREENWIDTH, height: dwBUTTONWIDTH*2))
        payMethodTableView.rowHeight = dwBUTTONWIDTH
        payMethodTableView.dataSource = self
        payMethodTableView.delegate = self
        payMethodTableView.bounces = false
        scrollView.addSubview(payMethodTableView)
        
        
        let lineLabel2 = UILabel(frame: CGRect(x: 15, y: payMethodTableView.frame.origin.y, width: dwDEVICESCREENWIDTH, height: 0.2))
        lineLabel2.backgroundColor = UIColor.lightGray
        scrollView.addSubview(lineLabel2)
        
        
    }
    
    
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return titleArray.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let cell = UITableViewCell(style: UITableViewCellStyle.value1, reuseIdentifier: "cellIdentifier")
        cell.selectionStyle = UITableViewCellSelectionStyle.none
        
        cell.textLabel?.text = titleArray[indexPath.row]
        
        if indexPath.row == 0 {
            cell.accessoryType = UITableViewCellAccessoryType.disclosureIndicator
            cell.detailTextLabel?.text = payMethodDict.object(forKey: "bankCard") as! String?
        }else if indexPath.row == 1 {
            
            cell.detailTextLabel?.text = payMethodDict.object(forKey: "userName") as! String?
        }
        
        return cell
        
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        if indexPath.row == 0 {
            self.bankListClick()
        }
    }
    
    func disMissView() {
        
        
        scrollView.frame = CGRect(x: 0, y: dwDEVICESCREENHEIGHT - dwSCROLLVIEWHEIGHT, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT)
        
        
        UIView.animate(withDuration: 0.3, animations: {
            self.alpha = 0.0
            self.scrollView.frame = CGRect(x: 0, y: dwDEVICESCREENHEIGHT, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT)
        }) { (finished) in
            self.scrollView.setContentOffset(CGPoint(x:0,y:0), animated: false)
            self.removeFromSuperview()
            self.scrollView.removeFromSuperview()
            
            
            NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "finishPay"), object: nil)
            NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "disableBackBtn"), object: nil)
            NotificationCenter.default.removeObserver(self, name: NSNotification.Name(rawValue: "bankCardSelected"), object: nil)
        }
        
        
    }
    
    func nextClick() {
        var dict = NSDictionary()
        dict = payMethodDict
        NotificationCenter.default.post(name: NSNotification.Name(rawValue: "bankInfoForPay"), object: nil, userInfo: dict as? [AnyHashable : Any] )//将银行卡信息传到DWManageData中，方便最后提交交易使用
        
        let offsetX = scrollView.bounds.size.width
        scrollView.setContentOffset(CGPoint(x:offsetX,y:0), animated: true)
        secondTitle.text = "输入数字支付密码"
        
        enterPasswordView = DWEnterPasswordView(frame: CGRect(x: dwDEVICESCREENWIDTH, y: dwBUTTONWIDTH, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT))
        enterPasswordView.payBankInfo = payMethodDict
        scrollView.addSubview(enterPasswordView)
        
    }
    
    func backClick() {
        scrollView.setContentOffset(CGPoint(x:0,y:0), animated: true)
        
        enterPasswordView.textField.resignFirstResponder()
        
    }
    
    //输入完密码后禁用返回键
    func disableBackBtn(){
        
        backBtn.isEnabled = false
        backBtn.adjustsImageWhenDisabled = false
        
    }
    
    
    /*
     回传选号的银行卡信息
     可将服务器报文直接转成字典传过来，也可以重新生成字典来传、
     记得修改-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath中的payMethodDict字典key值，以便正确显示银行卡和姓名
     */
    func bankCardSelected(notification:Notification){
        
        self.backClick()
        let userInfo = notification.userInfo
        
        
        let dict = userInfo as? Dictionary<String,String>
        payMethodDict = NSMutableDictionary(dictionary: dict!)
        payMethodTableView.reloadData()
        
    }
    
    func bankListClick(){
        
        let offsetX = scrollView.bounds.size.width
        scrollView.setContentOffset(CGPoint(x:offsetX,y:0), animated: true)
        secondTitle.text = "选择支付方式"
        
        let bankListView = DWBankListView(frame: CGRect(x: dwDEVICESCREENWIDTH, y: dwBUTTONWIDTH, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT))
        scrollView.addSubview(bankListView)
        
    }
    
    
    
    func showInView(view:UIView) {
        
        view.addSubview(self)
        view.addSubview(scrollView)
        
        scrollView.frame = CGRect(x: 0, y: dwDEVICESCREENHEIGHT, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT)
        
        UIView.animate(withDuration: 0.3) {
            
            self.alpha = 1.0
            self.backBtn.isEnabled = true;
            self.scrollView.frame = CGRect(x: 0, y: dwDEVICESCREENHEIGHT - dwSCROLLVIEWHEIGHT, width: dwDEVICESCREENWIDTH, height: dwSCROLLVIEWHEIGHT)
            self.getInitPayBankInfo()
            
            
            
            NotificationCenter.default.addObserver(self, selector: #selector(self.disMissView), name: NSNotification.Name(rawValue: "finishPay"), object: nil)//完成交易后，页面消失观察者
            NotificationCenter.default.addObserver(self, selector: #selector(self.disableBackBtn), name: NSNotification.Name(rawValue: "disableBackBtn"), object: nil)//输入密码，使返回按钮失效观察者
            NotificationCenter.default.addObserver(self, selector: #selector(self.bankCardSelected), name: NSNotification.Name(rawValue: "bankCardSelected"), object: nil)//选择完银行卡，返回传值观察者
            
            
        }
        
    }
    
    func getInitPayBankInfo() {
        let manageData = DWManageData()
        manageData.delegate = self
        manageData.getInitPayMethod()
    }
    
    /*
     代理传值获取到的默认银行卡信息
     可根据服务器返回信息自行修改
     记得修改-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath中的payMethodDict字典key值，以便正确显示银行卡和姓名
     */
    func passData(data: Any) {
        
        payMethodDict = data as! NSDictionary
        
        payMethodTableView.reloadData()
    }
    
    
}
